linq4j/src/main/java/org/apache/calcite/linq4j/tree/ConstantExpression.java

```diff
@@ -147,11 +147,15 @@ private static ExpressionWriter write(ExpressionWriter writer,
       try {
         final int scale = bigDecimal.scale();
         final long exact = bigDecimal.scaleByPowerOfTen(scale).longValueExact();
-        writer.append("new java.math.BigDecimal(").append(exact).append("L");
-        if (scale != 0) {
-          writer.append(", ").append(scale);
+        if (scale == 0) {
+          writer.append("new java.math.BigDecimal(")
+                .append(exact).append("L").append(")");
+        } else {
+          writer.append("java.math.BigDecimal.valueOf(")
+                .append(exact).append("L")
+                .append(", ").append(scale).append(")");
         }
```

Comment:
Maybe just unify this if else and use `java.math.BigDecimal.valueOf` for both scale=0 and scale!=0 case?

Per Java doc, `valueOf` is  the preferred way than BigDecimal constructor

Reply:
+1, it doesn't hurt to unify them.

Reply:
@amaliujia @hsyuan addressed, thanks for review and suggestion!

Reply:
LGTM